Framework/Parallel/test/EventParserTest.h

```diff
@@ -0,0 +1,464 @@
+#ifndef MANTID_PARALLEL_COLLECTIVESTEST_H_
+#define MANTID_PARALLEL_COLLECTIVESTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidParallel/IO/EventParser.h"
+
+#include <numeric>
+
+using namespace Mantid;
+using namespace Parallel::IO;
+using Mantid::Types::Core::DateAndTime;
+using Mantid::Types::Event::TofEvent;
+
+namespace detail {
+template <typename IndexType, typename TimeZeroType, typename TimeOffsetType>
+class FakeParserDataGenerator {
+public:
+  FakeParserDataGenerator(size_t numBanks, size_t pixelsPerBank,
+                          size_t numPulses) {
+    generateTestData(numBanks, pixelsPerBank, numPulses);
+  }
+
+  ~FakeParserDataGenerator() {
+    for (auto *list : m_event_lists)
+      delete list;
+  }
+
+  const std::vector<int32_t> &bankOffsets() const { return m_bank_offsets; }
+
+  const std::vector<IndexType> &eventIndex(size_t bank) const {
+    return m_event_indices[bank];
+  }
+
+  const std::vector<TimeZeroType> &eventTimeZero() { return m_event_time_zero; }
+
+  const std::vector<TimeOffsetType> &eventTimeOffset(size_t bank) const {
+    return m_event_time_offsets[bank];
+  }
+
+  const std::vector<int32_t> &eventId(size_t bank) const {
+    return m_event_ids[bank];
+  }
+
+  const std::vector<std::vector<TofEvent> *> &eventLists() const {
+    return m_event_lists;
+  }
+
+  LoadRange generateBasicRange(size_t bank) {
+    LoadRange range;
+    range.eventOffset = 0;
+    range.eventCount = m_event_ids[bank].size();
+    range.bankIndex = bank;
+
+    return range;
+  }
+
+  // TODO: Generate load ranges for different MPI ranks
+  // std::vector<LoadRange> generateMPIRanges(size_t bank, int rank,
+  //                                         int numRanks) {
+  //  return std::vector<LoadRange>{};
+  //}
+
+  static EventParser<IndexType, TimeZeroType, TimeOffsetType>
+  createSimpleParser(std::vector<std::vector<TofEvent> *> &eventLists,
```

Comment:
Don't understand why this creational method is on the generator. Doesn't make 100% sense to me.

Reply:
I did this so I could access these methods from the performance test. My initial idea was to create a parser based on the data generated. This will be modified shortly either way.